core/src/main/java/org/apache/iceberg/BaseMetastoreTableOperations.java

```diff
@@ -184,13 +184,17 @@ protected void refreshFromMetadataLocation(String newLocation, Predicate<Excepti
             "Table UUID does not match: current=%s != refreshed=%s", currentMetadata.uuid(), newUUID);
       }
 
-      this.currentMetadata = newMetadata.get();
+      this.currentMetadata = maybeValidateAndUpdateMetadata(newMetadata.get());
```

Comment:
I don't like this as a pattern. It feels like bad OOP. Can this be refactored? Or can `NessieCatalog` re-implement `refreshFromMetadataLocation`?